### ISSUE TITLE
[V3 Permissions] 3rd party checks required modifications.

### ISSUE THREAD

**Issue by mikeshardmind** [Cog-Creators/Red-DiscordBot#1758]
With the current model for hooking third party checks into the permissions cog, we need permissions to be loaded first and not be reloaded. This was a design oversight on my part. 

I'll be postponing my other WIP stuff to resolve this in a satisfactory manner prior to b16, but this will require changes to how other checks which want to integrate into the permissions cog rather than operate separately need to function.

For anyone currently designing a 3rd party cog which uses these hooks: 
After the intended changes, any function which is an attribute of your cog should be possible to use as an integrated check. Each cog will get 1 before and 1 after check with expected names.

`__red_permissions_before` and `__red_permissions_after`
These can call your already existing functions if needed, and will receive the same arguments the current which current checks already do.

**Comment by mikeshardmind** [comment]
After these changes, cogs can have the following methods (async optional) 

```py
async def __red_permissions_before(ctx: commands.Context, *, level: str)
```

```py
async def __red_permissions_after(ctx: commands.Context, *, level: str)
```
These methods should return `None` if not interfering with the normal check operation, `True` if it should run, `False` if it should not.

More detailed documentation for this will be available at a later date, but the original API was also undocumented.

**Comment by tekulvw** [comment]
Implemented with #1760
